test/sys/test_socket.rs

```diff
@@ -301,14 +301,36 @@ fn test_scm_credentials() {
 /// Ensure that we can send `SCM_CREDENTIALS` and `SCM_RIGHTS` with a single
 /// `sendmsg` call.
 #[cfg(any(target_os = "android", target_os = "linux"))]
+// qemu's handling of multiple cmsgs is bugged, ignore tests on non-x86-64
```

Comment:
Can you add an upstream bug link?  Your C test case should make it easy to report.